cmd/dpbindic/dpbindic.man
.\".ds LB /usr/lib/canna
.ds LB /usr/local/canna/lib
.TH DPBINDIC 1
.SH NAME
dpbindic \- Convert a binary-form dictionary into a text-form dictionary
.SH SYNOPSYS
.B dpbindic
[
.B \-xiu
[
.I frequency
]
]
.I "binary-file"
[
.I "text-file"
]
.SH DESCRIPTION
.I dpbindic
outputs the file information of the binary-form dictionary file specified in
.I "binary-file"\fP.
At this time, the word information of the dictionary can be output in text
form to the standard output.  To do so, use
.I "test-file"
to specify the text-form dictionary used as the source of binary-form
dictionary file.  If this specification is omitted, the text
dictionary file information in the binary dictionary file will be
output.  The standard grammar file name is
\*(LB/dic/hyoujun.gram. It will be used if the grammar file
name specification is omitted.  The output format of word information
data is specified using an option.
.SH OPTIONS
.TP
.B \-x
Outputs the data without using omission symbol @, which is used when the initial word represents the reading.
.TP
.B \-i
Replaces the reading and word for output.
.TP
.B \-u
Outputs the candidates used in conversion.  Outputs all candidates having 
.I frequency
or more.  If
.I frequency
is omitted, all candidates having frequency 1 will be output.
.SH EXAMPLES
.PP
(1) If the text-form dictionary file name is omitted:
.nf
.B
%dibindic iroha.cbd
.B
(Text dictionary file name = Directory size + Word size, packed)
.B
iroha.swd = 2985 + 5306 pak a4
.B
iroha.mwd = 36276 + 113139 pak a4
.PP
(2) If the text-form dictionary file name iroha.mwd is specified:
.nf
.B
%dpbindic iroha.cbd iroha.mwd
.B
(Text dictionary file name = Directory size + Word size, packed)
.B
iroha.mwd = 36276 + 113139 pak a4

.SH SEE ALSO
mkbindic(1), dicar(1)

.\" Copyright 1994 NEC Corporation, Tokyo, Japan.
.\"
.\" Permission to use, copy, modify, distribute and sell this software
.\" and its documentation for any purpose is hereby granted without
.\" fee, provided that the above copyright notice appear in all copies
.\" and that both that copyright notice and this permission notice
.\" appear in supporting documentation, and that the name of NEC
.\" Corporation not be used in advertising or publicity pertaining to
.\" distribution of the software without specific, written prior
.\" permission.  NEC Corporation makes no representations about the
.\" suitability of this software for any purpose.  It is provided "as
.\" is" without express or implied warranty.
.\"
.\" NEC CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN 
.\" NO EVENT SHALL NEC CORPORATION BE LIABLE FOR ANY SPECIAL, INDIRECT OR
.\" CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF 
.\" USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR 
.\" OTHER TORTUOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR 
.\" PERFORMANCE OF THIS SOFTWARE. 
.\"
.\" $Id: dpbindic.man,v 1.2 2003/01/15 13:41:02 aida_s Exp $


ChangeLog
2004-05-21 AIDA Shinra <shinra@j10n.org>

	* Canna 3.7p3 release.
	* configure.ac: bump to 3.7p3
	* canna/patchlevel.h: ditto
	* CHANGES.jp: updated

2004-05-21 AIDA Shinra <shinra@j10n.org>

	* confwrapper: "." bashism. Many shells don't searches the current
	  directory.
	* Imakefile: $(CC) -> env $(CC) in case $(CC) is
	  "LD_LIBRARY_PATH=/usr/X11R6/lib cc".
	* canuum/Imakefile: ditto

2004-05-19 AIDA Shinra <shinra@j10n.org>

	* Canna 3.7p2 release.
	* configure.ac: bump to 3.7p2
	* canna/patchlevel.h: ditto

2004-05-05 AIDA Shinra <shinra@j10n.org>

	* CHANGES.jp: fix a typo

2004-05-05 AIDA Shinra <shinra@j10n.org>

	* COPYRIGHT: added
	* CHANGES.jp: updated for 3.7p2
	* RKCCONF.jp: warn about future change

2004-05-05 AIDA Shinra <shinra@j10n.org>

	* Merge Pubdic+-fix01 to make the license of pubdic more clear.
	* dic/ideo/pubdic/pod.c: Remove #ifndef POD_WCHAR part.
	  Make casts more proper. Add 'vu' conversion though it is not needed
	  for now.
	* dic/ideo/pubdic/COPYRIGHT: added
	* dic/ideo/pubdic/VERSION: added

2004-04-27 AIDA Shinra <shinra@j10n.org>

	* VENDOR_DEFINES which comes from X.Org's imake contains single
	  quotes.
	* confwrapper: generate cwenv.sh first and then run configure
	* Imakefile: ditto
	* canuum/Imakefile: ditto
	* confwrapper.sub: removed
	* Canna.conf.dist: DLLs need INSTBINFLAGS instead of INSTLIBFLAGS

2004-04-27 AIDA Shinra <shinra@j10n.org>

	* When delay-connect is t and KC_FINALIZE is used uilib gets into
	  inconsistent state.
	* lib/canna/commondata.c(mountnottry): added
	* lib/canna/globnames: added mountnottry
	* lib/canna/henkan.c(KanjiInit): mountnottry is now global symbol
	  (KanjiFin): clear some variables

2004-04-27 AIDA Shinra <shinra@j10n.org>

	* Merge trivial fixes (uilib, doc)
	* lib/canna/obind.c(owcLookupKanji2): buffer_return is not char *
	  but wchar_t *.
	  (cannawc32): use canna_uint32_t
	  (cannawc16): use canna_uint16_t
	* lib/canna/RKroma.c(RkMapPhonogram): KPDIC&&PTDIC -> KPDIC||PTDIC
	* [Canna-dev 297]: sentou 1moji sentaku bug
	* lib/canna/henkan.c(TbChooseChar): last character's attribute needs
	  SENTOU flag
	* [Canna-dev 290]: Documents
	* doc/man/guide/tex/func.tex: Now renbun-continue is t by default.
	* dic/ideo/grammar/gram.code: correct a comment
	* [Canna-dev 291]: Protocol definition is inconsistent with
	  implementation.
	* doc/intern/protocol.tex: corrected
	* [Canna:05898]: canna-parse is broken
	* lib/canna/lisp.c(Lread): stop fclose(NULL)

2004-04-27 AIDA Shinra <shinra@j10n.org>

	* Merge trivial fixes (canuum, RKC and server)
	* canuum/canuum.jmn: Description about -[SJUsju] options were
	  misleading or wrong.
	* canuum/jhlp.c(chld_handler): cygwin has union wait but WIF*() macros
	  do not work if union wait is given
	* canuum/termcap.c(sr_set): check %p though it exists only in terminfo
	* lib/RKC/conf.c(read_pipe_with_errors): bug fix in no_exitstatus
	  case (thanks to nanashi san)
	  (Lexer_next): an extra memchr() caused a crash
 	* lib/RKC/convert.c(RkcSendERequest): When communication fails, set
 	  errno to EPIPE and close the socket in any case.
 	* lib/RKC/wconvert.c(RkcSendWRequest): ditto
	* server/misc.c(mysignal): added to prevent syscalls from restarting
	* server/Imakefile: SIGNAL_DEFINES
	* server/comm.c(ClientBuf): count temporary failures and close
	  connection if too many errors happen
	  (ClientBuf_recv): save errno before Dmsg
	  (ClientBuf_send): ditto
	  (EventMgr_run): ditto

2004-01-02 AIDA Shinra <shinra@j10n.org>

	* Canna 3.7p1 release.
	* CHANGES.jp: updated
	* canna/patchlevel.h: bump to 3.7p1

2003-12-29 AIDA Shinra <shinra@j10n.org>

	* Canna.conf.dist(sharedLibExtension): HPArchitecture is also defined
	  on Linux with hppa. Make sure the system is really HP-UX.
	  ([Canna-dev 280])
	* confwrapper: CC="$ccadd $CC" is foolish ([Canna-dev 281])

2003-12-28 AIDA Shinra <shinra@j10n.org>

	* Canna.conf.dist(SHLIBLDFLAGS): Assign "-G -z text" on SVR4
	  because Solaris appends "-M mapfile" and therefore we need to
	  generate mapfile.
	* server/wconvert.c(ExtensionVector): remove "static"
	* CHANGES.jp: make mention of glibc

2003-12-28 AIDA Shinra <shinra@j10n.org>

	* Sync with HEAD. All changes were fixes of bugs or portability
	  problems, therefore I merged all of them.
	* CHANGES.jp: forgot IPv6 feature
	* Canna.conf.dist(CppSedMagic): Remove #pragma inserted by the
	  default gcc-3.3 that ships with MacOS X 10.3. Recent snapshot
	  of XFree86 specifies /usr/bin/cpp3 as CppCmd so we don't need
	  redefine CppSedMagic.
	* Imakefile: Use confwrapper. #undef malloc in cannaconf.h.
	  Fix OS2Architecture.
	* configure.ac: add proper arguments of AC_INIT and RCSID
	* confwrapper: added to remove something like ` ` in CPPFLAGS
	  and others
	* confwrapper.sub: ditto
	* mkrelease.sh: always use autoconf 2.59
	* canuum/Imakefile: use confwrapper
	* dic/ideo/pubdic/Imakefile: link libRKindep
	* lib/RKC/conf.c(Lexer_next): skip #pragma
	  (host_str_defaults): put a dummy member
	  (top_num_defaults): ditto
	* lib/RKindep/cfuncs.c: #ifndef HAVE_MALLOC -> #if !HAVE_MALLOC
	* lib/RKindep/cfuncs.h: Newer autoconf defines HAVE_MALLOC to 0 but
	  older undefines it. Accept both.
	* server/convert.c: Forward declaration of file scope array is not
	  confirm to the standard. Fixed. Added some #ifdef USE_EUC_PROTOCOL.
	* server/wconvert.c: ditto

2003-12-13 AIDA Shinra <shinra@j10n.org>

	* CHANGES.jp: modify description of old "security bug"


2003-12-12 AIDA Shinra <shinra@j10n.org>

	* Canna 3.7 release.
	* CHANGES.jp: updated
	* canna/patchlevel.h: bump to 3.7

2003-12-12 AIDA Shinra <shinra@j10n.org>

	* dic/ideo/*: Sync with DIC_WORK branch.

2003-12-07 AIDA Shinra <shinra@j10n.org>

	* canna/widedef.h: MacOS X 10.3 support
	* canna/jrkanji.h: define CANNA_JR_BEEP_FUNC_DECLARED

2003-10-12 AIDA Shinra <shinra@j10n.org>

	* Imakefile: s/CANNA_PURE_CFLAGS/PURE_CFLAGS/
	* configure.ac: check inet_aton
	* server/misc.c(GetAddrListFromName): use inet_addr if inet_aton is

cmd/dpbindic/dpbindic.cpp
XCOMM!/bin/sh
XCOMM Copyright 1992 NEC Corporation, Tokyo, Japan.
XCOMM
XCOMM Permission to use, copy, modify, distribute and sell this software
XCOMM and its documentation for any purpose is hereby granted without
XCOMM fee, provided that the above copyright notice appear in all copies
XCOMM and that both that copyright notice and this permission notice
XCOMM appear in supporting documentation, and that the name of NEC
XCOMM Corporation not be used in advertising or publicity pertaining to
XCOMM distribution of the software without specific, written prior
XCOMM permission.  NEC Corporation makes no representations about the
XCOMM suitability of this software for any purpose.  It is provided 
XCOMM "as is" without express or implied warranty.
XCOMM
XCOMM NEC CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
XCOMM INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN 
XCOMM NO EVENT SHALL NEC CORPORATION BE LIABLE FOR ANY SPECIAL, INDIRECT OR
XCOMM CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF 
XCOMM USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR 
XCOMM OTHER TORTUOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR 
XCOMM PERFORMANCE OF THIS SOFTWARE. 

XCOMM $Id: dpbindic.cpp,v 1.1.1.1 2002/10/19 08:27:34 aida_s Exp $
#include "cannaconf.h"
PATH=CANNABINDIR:$PATH:/bin:/usr/bin:/etc:/usr/etc:/usr/nec/bin:/usr/ucb
export PATH

echo dpxdic $* 1>&2
dpxdic $*
